src/metadata/duplicate_obs/script.py

```diff
@@ -38,9 +39,19 @@ def setup_logger():
 mdata = read_h5mu(par['input'])
 adata = mdata.mod[par['modality']]
 
-if not par["input_obs_key"] == par["output_obs_key"]:
-    logger.info(f"Copying .obs key {par['input_obs_key']} to {par['output_obs_key']}")
-    adata.obs[par["output_obs_key"]] = adata.obs[par["input_obs_key"]].copy()
+if not par["output_obs_key"] in adata.obs:
+    if par["input_obs_key"]:
+        logger.info(f"Copying .obs key {par['input_obs_key']} to {par['output_obs_key']}")
+        adata.obs[par["output_obs_key"]] = adata.obs[par["input_obs_key"]].copy()
+    else:
+        logger.info(f"Copying .obs index to {par['output_obs_key']}")
+        adata.obs[par["output_obs_key"]] = adata.obs.index.copy()
 
+else:
+    if par["disable_raise_on_identical_keys"]:
+        logger.warning(f"--output_obs_key already exists: `{par['output_obs_key']}`. Data can not be duplicated.")
+    else:
+        raise ValueError(f"--output_obs_key already exists: `{par['output_obs_key']}`. Data can not be duplicated.")
```

Comment:
Maybe its a good idea to add an `override` parameter instead of `disable_raise_on_identical_keys`? That way the use can force the component to function even when the key already exists? AFAIK, with the proposed implementation, the component can never copy with overlapping keys?